### ISSUE TITLE
icons missing in app launcher

### ISSUE THREAD

**Issue by mbtuckersimm** [regolith-linux/ilia#51]
I just upgraded to Regolith 2.1, and I see no app icons in the ilia app launcher. This happens both when I open the launcher using `Super+Space` and when I launch it from the shell with

```
ilia -p apps
```

I checked the setting `org.regolith-linux.ilia.icon-size` in dconf-editor and it is set to the default value of 32, so I don't think that's the problem.

**Comment by kgilmer** [comment]
Hi @mbtuckersimm , what variant of Regolith are you using (deb repo or iso) and what OS and version are you running? (debian, ubuntu, 2x.xx..)

**Comment by mbtuckersimm** [comment]
@kgilmer Sorry for omitting that info.

I'm using the deb repo, running Linux Mint 21 (sitting on top of Ubuntu 22.04). Please let me know if there's any further diagnostic info you need.

**Comment by mbtuckersimm** [comment]
Further info: the icons do appear the first time I open the launcher after logging in, but not after that. So maybe some kind of config variable is getting overwritten?

**Comment by kgilmer** [comment]
Hi @mbtuckersimm , thanks for the additional info.  Another user has reported this behavior in Ubuntu 22.10.  If you run ilia from a terminal, do you see any warning messages printed that may be related to the icons?  

[Ilia loads icons](https://github.com/regolith-linux/ilia/blob/main/src/apps/DesktopAppPage.vala#L253) from GTK, unsure of what in particular ilia would be doing wrong here..

**Comment by cjthompson** [comment]
I just noticed this as well. It started within the last week or so for me. I don't see icons anymore.

**Comment by mbtuckersimm** [comment]
OK, further interesting information: if I run `ilia -a` to open ilia with all of the panels, icons _do_ appear in the Notifications pane (always, it seems) and they _sometimes_ appear in the Windows pane.

Also, I do get an error when I run `ilia -a`:

```
matthew@ultima-thule /home/matthew
> ilia -a

** (ilia:743083): CRITICAL **: 14:41:49.540: ilia_load_icon_from_info: assertion 'app_info != NULL' failed
```

**Comment by mbtuckersimm** [comment]
> Also, I do get an error when I run `ilia -a`:

I get this same error when I run `ilia -p windows` but _not_ when I run `ilia -p apps`.

**Comment by kgilmer** [comment]
Nice, this is progress. @mbtuckersimm to clarify, does the error and "no icons" always occur together?

**Comment by mbtuckersimm** [comment]
@kgilmer Here's a summary of the behavior I'm seeing:

- `ilia -p windows`: I get the error and I don't see any icons
- `ilia -p apps`: no error and no icons
- `ilia -a`: I get the error, there are no icons in the apps or windows pane but there _are_ icons in the notifications pane

However, this behavior is not 100% consistent. Once in a while, if I run `ilia -a`, there is no error and the icons _do_ appear in the windows pane (and in this case they still do appear in the notifications pane too, but they still don't appear in the apps pane).

So it appears that the error is correlated with the icons not appearing _in the windows pane only_. The error doesn't seem to manifest when loading the apps list, even though the icons don't appear there.

**Comment by kgilmer** [comment]
@mbtuckersimm I have made some changes to desktop app logic since 2.1, and I'm wondering if you could check and see if the behavior you're seeing has changed.  You could either compile ilia from src (on `ubuntu/jammy` branch for Ubuntu versions jammy or greater) using instructions in [readme](https://github.com/regolith-linux/ilia/blob/main/README.md#build), although you may not want to install all the build dependencies, or install a binary package directly from the apt repo:

Ubuntu Kinetic URL:  https://regolith-desktop.org/unstable-ubuntu-kinetic-amd64/pool/main/i/ilia/ilia_0.12.26-jammy-1regolith_amd64.deb

Happy to provide URLs for other versions of Ubuntu or Debian depending on your system..

**Comment by mbtuckersimm** [comment]
@kgilmer Thanks so much! I will try the .deb first. I currently have the following ilia-related packages installed:

- ilia
- regolith-i3-ilia
- regolith-i3-rofication-ilia

Do I need to uninstall any of them before installing the .deb you linked to?

**Comment by kgilmer** [comment]
I don't think so.  I think `ilia` will just upgrade, as the version in this deb will be greater than your installed version.  Note what you have installed so that you can [easily downgrade/revert](https://askubuntu.com/questions/138284/how-to-downgrade-a-package-via-apt-get) if something goes wrong.

**Comment by msp-matt** [comment]
OK, so I think I figured out what's going on: it's a config problem of some sort. Here's the behavior that I saw when I installed the new version from the deb file:

- `ilia -p apps`: no errors and icons did appear
- `ilia -p windows`: now the pane is empty (not only no icons but no apps listed either), and I get this error:
```
> ilia -p windows
Failed to read or parse window tree from i3: Connection refused
```

- `ilia -p keybindings`: no keybindings shown, and I see this error:
```
> ilia -p keybindings
Failed to read config from i3: Connection refused
```

- `ilia -a`: apps pane is ok (icons appear) but windows list and keybindings list are both empty, and I get both errors that I posted above, plus an additional one:
```
> ilia -a
Failed to read config from i3: Connection refused
Failed to read or parse window tree from i3: Connection refused

(ilia:908083): Gtk-CRITICAL **: 15:38:19.588: gtk_widget_event: assertion 'WIDGET_REALIZED_FOR_EVENT (widget, event)' failed
```

So you can see that there seems to be some config problem. I've been playing around with the regolith/i3 config recently trying to understand how things work and how to tweak settings, and what happened was that I had a config directory
```
~/.config/regolith2/i3/config.d
```
with no files in it.

When I removed the empty directory `config.d` and logged out and back in again (restarting i3 didn't seem to be enough), all the problems went away: there were no errors, icons appeared in the ilia apps list, and the windows list and keybindings viewer both worked normally.

I didn't downgrade back to the previous version to check if the empty config directory was the cause of the problems with that version also, but I could do it if that would be helpful.

Incidentally, is there any way to see what config files i3 is loading and in what order? It would be really helpful to be able to log that, as I'm a bit confused about what happens at what times.

**Comment by mbtuckersimm** [comment]
Oops, https://github.com/regolith-linux/ilia/issues/51#issuecomment-1321240787 is also me. Forgot I was using my work browser, logged in to my work account. It's not a mysterious other person with my same profile pic.

**Comment by kgilmer** [comment]
> ilia -p apps: no errors and icons did appear
Yay :tada: 

>  i3: Connection refused

In all cases in which you saw this behavior, it would appear that i3 isn't responding on it's socket.  I don't see this as being a ilia bug. Unsure of what would cause that but I wonder if logging back in or refreshing i3 would fix it.

> with no files in it.

Very odd.  I'll have to look at the code again to see what an empty directory might do.  But, ilia's icon loading and i3 config..I'm struggling to think of how they could be linked.

> Incidentally, is there any way to see what config files i3 is loading and in what order?

Yes: `i3 --moreversion`

Thanks for all the details and taking the time to setup the test version.  It really helps me in making a call about finalizing 2.2.

**Comment by mbtuckersimm** [comment]
Thanks so much and glad I could help out a little. I really appreciate your assistance here and all the contributors' work on this project in general.

**Comment by krishnakumarg1984** [comment]
I am the other user who reported this issue on the slack forum. `ilia -a` does not help in my case, and the icons are still missing.

I get the following error:
```
$ ilia -a
select_path

(ilia:506161): Gtk-CRITICAL **: 19:49:13.655: gtk_widget_event: assertion 'WIDGET_REALIZED_FOR_EVENT (widget, event)' failed
```

**Comment by kgilmer** [comment]
Hi @krishnakumarg1984 , did you update `ilia` from the other repo as discussed above, or are you using the version that shipped in version 2.1?

**Comment by kgilmer** [comment]
I have reproduced the issue on 22.10 from the ilia version in the `unstable` repo.  Fix in process.

**Comment by kgilmer** [comment]
The bug returned in debian testing just now, ripping out more `async` stuff..

**Comment by kgilmer** [comment]
This should be fixed in 2.2.  Kindly reopen if anyone is still seeing this behavior
